lib/open-uri-cached-archive.rb

```diff
@@ -1,5 +1,43 @@
 require 'open-uri-cached-archive/version'
+require 'yaml'
+require 'stringio'
+require 'open-uri'
 
 class OpenUriCachedArchive
-  # Your code goes here...
+  class CachedResponse
+    def initialize(meta_file_path)
+      @meta = YAML.load_file(meta_file_path)
+      @body = File.read(meta_file_path.gsub(/\.meta$/, ''))
+      @status = meta.delete(:status)
+      @base_uri = meta.delete(:base_uri)
+      meta.delete(:content_type)
+    end
+
+    def open_uri_response
+      StringIO.new(body).tap do |response|
+        OpenURI::Meta.init(response)
+        meta.each { |k, v| response.meta_add_field(k, v) }
+        response.status = status
+        response.base_uri = base_uri
+      end
+    end
+
+    private
+
+    attr_reader :meta, :body, :status, :base_uri
+  end
+
+  def initialize(path)
+    @path = path
+  end
+
+  def responses
+    Dir.glob(File.join(path, '**', '*.meta')).map do |meta_file_path|
```

Comment:
I didn't even know about `Pathname.glob`, what a handy method! Turns out there is a `Pathname#sub_ext` method, which will replace the extension of the current pathname. Passing that an empty string gives us the desired behaviour. Done in https://github.com/everypolitician/open-uri-cached-archive/pull/3/commits/be62623619fde7cfb5c59a2ced51693a347cb0c0